__test__/ship.test.js

```diff
@@ -0,0 +1,83 @@
+const Ship = require('../src/ship')
+const Port = require('../src/port')
+const Itinerary = require('../src/itinerary')
+
+describe('ship', () => {
+
+    describe('has ports and itinerarys', () => {
+        let ship
+        let amsterdam
+        let dover
+        let itinerary 
+
+        beforeEach(() => {
+        amsterdam = {
+            addShip: jest.fn(),
+            removeShip: jest.fn(),
+            name: 'Amsterdam',
+            ships: []
+        }
+        
+        dover = {
+            addShip: jest.fn(),
+            removeShip: jest.fn(),
+            name: 'Dover',
+            ships: []
+        }
+
+        itinerary = {
+            ports: [amsterdam, dover]
+        }
+
+        ship = new Ship(itinerary)
+
+        })
+        
+        it('returns an object', () => {
+            expect(ship).toBeInstanceOf(Object)
+        })
+
+        it('it should have a starting port', () => {
+
+             expect(ship.currentPort).toBe(amsterdam)
+        })
+
+        it('can set sail', () => {
+    
+            ship.setSail()
+          
+            expect(ship.currentPort).toBeFalsy()
```

Comment:
I would have written this test to be `expect(ship.currentPort).toEqual(null)` as it's more explicit with the value currentPort is being set to